Source/DafnyCore/Resolver/PreType/PreTypeResolve.cs

```diff
@@ -520,17 +539,22 @@ void AddComparableConstraint(PreType a, PreType b, IOrigin tok, bool allowBaseTy
       Constraints.AddGuardedConstraint(() => ApproximateComparableConstraints(a, b, tok, allowBaseTypeCast,
         "(Duplicate error message) " + errorMessage(), false));
       if (!allowBaseTypeCast) {
-        // The "comparable types" constraint may be useful as a bound if nothing else is known about a proxy. 
-        if (a.Normalize() is PreTypeProxy aPreTypeProxy) {
-          Constraints.AddCompatibleBounds(aPreTypeProxy, b);
-        }
-        if (b.Normalize() is PreTypeProxy bPreTypeProxy) {
-          Constraints.AddCompatibleBounds(bPreTypeProxy, a);
-        }
+        AddComparableTypesDefault(a, b);
       }
       Constraints.AddConfirmation(tok, () => CheckComparableTypes(a, b, allowBaseTypeCast), errorMessage);
     }
 
+    private void AddComparableTypesDefault(PreType a, PreType b) {
```

Comment:
Would be good to have documentation on what practical Dafny code this is helping with